src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -0,0 +1,176 @@
+				if !strings.Contains(topic, "/") {
+					topics[i] = "kafka/" + topic
+				}
+			}
+			for _, partition := range partitions {
+				if len(topics) > 0 { // foo -p 0:1,2,3
+					p, replicas, err := parsePartitionReplicas(partition)
+					out.MaybeDie(err, "unable to execute the request: %v\n", err)
+					for _, nt := range topics {
+						err = cl.MoveReplicas(cmd.Context(), nt+"/"+p, replicas)
+						out.MaybeDie(err, "unable to move partition replicas: %v\n", err)
+					}
+				} else { // -p foo/0:1,2,3
+					ntp, replicas, err := parseTopicPartitionReplicas(partition)
+					out.MaybeDie(err, "unable to execute the request: %v\n", err)
+					err = cl.MoveReplicas(cmd.Context(), ntp, replicas)
+					out.MaybeDie(err, "unable to move partition replicas: %v\n", err)
+				}
+			}
+			fmt.Print("Successfully submitted the request(s). Check the status with 'rpk cluster partitions move-status' or see the new assignments with 'rpk topic describe -p TOPIC'.\n")
+		},
+	}
+	cmd.Flags().StringArrayVarP(&partitions, "partition", "p", nil, "Partitions to move and new replica locations (repeatable)")
+	cmd.MarkFlagRequired("partition")
+	return cmd
+}
+
+// Expected input format is '0:1,2,3' or '1:1-0,2-0,3-0'
+// If '0:1,2,3' is passed, returns
+//
+// - '0'
+// - [{NodeID:1},{NodeID:2},{NodeID:3}]
+//
+// If '1:1-0,2-0,3-0' is passed, returns
+//
+// - '1'
+// - [{NodeID:1,Core:0},{NodeID:2,Core:0},{NodeID:3,Core:0}].
+func parsePartitionReplicas(partition string) (string, []adminapi.Replica, error) {
+	var replicas []adminapi.Replica
+	pr := strings.Split(partition, ":")
+	if len(pr) != 2 {
+		return "", nil, fmt.Errorf("invalid format for %s", partition)
+	}
+	if strings.Contains(pr[0], "/") {
+		return "", nil, fmt.Errorf("invalid format for %s; unexpected \"/\"", partition)
+	}
+	replicasTmp := strings.Split(pr[1], ",")
+	for _, replicaCore := range replicasTmp {
+		nc := strings.Split(replicaCore, "-")
+		if len(nc) > 1 {
+			nodeID, err := strconv.Atoi(nc[0])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			core, err := strconv.Atoi(nc[1])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[1], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+				Core:   core,
+			})
+		} else {
+			nodeID, err := strconv.Atoi(replicaCore)
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+			})
+		}
+	}
+	return pr[0], replicas, nil
+}
+
+// Expected input format is 'foo/0:1,2,3' or 'redpanda/controller/0:1-0,2-0,3-0'
+// If 'foo/0:1,2,3' is passed, returns
+//
+// - 'kafka/foo/0'
+// - [{NodeID:1,},{NodeID:2},{NodeID:3}]
+//
+// If 'redpanda/controller/0:1-0,2-0,3-0' is passed, returns
+//
+// - 'redpanda/controller/0'
+// - [{NodeID:1,Core:0},{NodeID:2,Core:0},{NodeID:3,Core:0}].
```

Comment:
Doesn't this mean to move all these partitions to core 0? Do we need the core do be a nil pointer instead?

Reply:
Thanks. I find core is 0 by default which is not good. I want to leave the assigned core as is when it's not specified explicitly. If `Core` is nil the api request fails, therefore I may need to marshal the json before passing it to the api.

Reply:
I see it's done via `func (nc nc) String() string { ... }`  in your example.

Reply:
Interesting, so does that mean we have high risk of moving all partitions to core 0? Isn't that bad?

Reply:
Upon discussing with the core team, we're leaning towards this direction:

* Currently the 'core' field is mandatory in the API.
  * Redpanda core team will make a change on it being optional and let the internal partition allocator decide the target shard if it's not specified.
* So far, RPK does these:
  * If 'core' is specified, use it.
  * If 'core' is not specified for a new node, find a new shard in a random fashion.
  * If 'core' is not specified for an existing node, use the current shard as is.

Reply:
Fixed in https://github.com/redpanda-data/redpanda/commit/1595272ed458ac47621cdbcad4c550c99140d622